src/app/features/courses/courses.component.ts

```diff
@@ -13,58 +14,60 @@ import { Subject, takeUntil } from 'rxjs';
 })
 export class CoursesComponent implements OnInit, OnDestroy {
   private destroy$: Subject<boolean> = new Subject();
-  courses!: Course[];
-  selectedCourse!: Course | null;
+  courses$ = this.coursesFacade.allCourses$;
+  selectedCourse$: Observable<Course | null>;
+  course!: Course | null;
   isAdmin!: boolean;
 
   protected readonly ButtonText = ButtonText;
 
   constructor(
     private coursesStoreService: CoursesStoreService,
     private userService: UserStoreService,
-    private router: Router
-  ) {}
+    private router: Router,
+    private route: ActivatedRoute,
+    private coursesFacade: CoursesStateFacade
+  ) {
+    this.selectedCourse$ = this.coursesFacade.course$;
```

Comment:
Since you are assigning `courses$` some value up-front, I guess you can do the same for `selectedCourse$`

Reply:
yeah, i changed it